### ISSUE TITLE
npm

### ISSUE THREAD

**Issue by SebastianBaltes** [manthrax/THREE-CSGMesh#9]
Thanks alot for this great library! Do you have any plans to publish it on npm? This would make it easier to use it.

**Comment by tianBreznik** [comment]
Hi! I'm struggling to use this package without npm and I'm also not finding much resources online on how to best include github packages in three.js projects. Could you perhaps point me in the right direction? As you might be thinking already, I'm pretty desperate...

**Comment by hamidb80** [comment]
there is a npm package called [three-csg-ts](https://www.npmjs.com/package/three-csg-ts) that packaged awesome work of `manthrax`  and added types to it

**Comment by trusktr** [comment]
TLDR: It is totally installable via `npm`, without it having to actually be published on npmjs.com, by installing it from github.

To do so, just fork the repo, then add this `package.json ` file to it:

```json
{
  "name": "three-csg",
  "version": "0.0.0",
}
```

then, install it like this:

```sh
npm install github-username/THREE-CSGMesh
```

and that will automatically install it in `node_modules/three-csg` by pulling it from https://github-username/THREE-CSGMesh. Just make sure to replace `github-username` with your actual GitHub username.

If you want to install a particular commit or branch, you can also specify that after a hash symbol:

```sh
npm install github-username/THREE-CSGMesh#d82h48g
# or
npm install github-username/THREE-CSGMesh#some-branch
```

Then import it in your code:

```js
import CSG from 'three-csg/three-scg.js'
```

If you are familiar enough with `package.json` configuration, you can even tweak it so that the `import` statement will be shorter:

```js
import CSG from 'three-csg'
```

but I will leave that as an exercise.

**Comment by manthrax** [comment]
Thanks for the great info, @trusktr!
